tests/python/pants_test/engine/test_fs.py

```diff
@@ -357,6 +357,23 @@ def test_merge_directories(self):
 
       self.assertEquals(both_snapshot.directory_digest, both_merged)
 
+  def test_materialize_directories(self):
+    with temporary_dir() as temp_dir:
+      dir_path = os.path.join(temp_dir, "roland")
+      digest = DirectoryDigest(
```

Comment:
Does anything guarantee that this digest is already in the store? If not, should probably copy the body of `test_snapshot_from_outside_buildroot` into this test (or a helper function) and run it, to prime the store.

(I'm actually not sure how this test is working right now... Oh... I know how... We don't create custom store paths for tests, it will just always use `${HOME}/.cache/pants/lmdb_store`, right? So I guess if you delete that directory, and run this test in isolation, it will probably fail?)

Reply:
Speaking of custom store paths, Is there an issue for adding isolated lmdb paths to tests? Would that make sense to do?

Reply:
> (I'm actually not sure how this test is working right now... Oh... I know how... We don't create custom store paths for tests, it will just always use ${HOME}/.cache/pants/lmdb_store, right? So I guess if you delete that directory, and run this test in isolation, it will probably fail?)

I can confirm that the test run on its own failed when I wiped `~/.cache/pants/lmdb_store`, but passed when the store was primed with the digest.

Reply:
There is now! https://github.com/pantsbuild/pants/issues/6038